relayer/pkg/chainlink/ocr2/client.go

```diff
@@ -143,27 +167,48 @@ func (c *Client) ConfigFromEventAt(ctx context.Context, address string, blockNum
 
 			m, err := json.Marshal(event)
 			if err != nil {
-				return cc, errors.Wrap(err, "couldn't marshal event")
+				return eventsAsFeltArrs, errors.Wrap(err, "couldn't marshal event")
 			}
 
 			err = json.Unmarshal(m, &decodedEvent)
 			if err != nil {
-				return cc, errors.Wrap(err, "couldn't unmarshal event")
+				return eventsAsFeltArrs, errors.Wrap(err, "couldn't unmarshal event")
 			}
 
-			if starknet.IsEventFromContract(&decodedEvent, address, "config_set") {
-				config, err := ParseConfigSetEvent(decodedEvent.Data)
-				if err != nil {
-					return cc, errors.Wrap(err, "couldn't parse config event")
-				}
-
-				return ContractConfig{
-					Config:      config,
-					ConfigBlock: blockNum,
-				}, nil
+			if starknet.IsEventFromContract(&decodedEvent, address, eventType) {
+				eventsAsFeltArrs = append(eventsAsFeltArrs, decodedEvent.Data)
```

Comment:
Could you please explain the purpose of this change?
I think it's guaranteed to have one or none config event per block, because otherwise we would have a design issue  since we refer to config by block number.

Reply:
Alright. Let's maybe find a better name for it, something like `fetchEventsFromBlock`?
And I would leave a comment at places where you use first array element (or even throw an error if there are more than one event) for better readability.

Reply:
Excellent ideas!

Reply:
Done in https://github.com/smartcontractkit/chainlink-starknet/pull/103/commits/53ab00621edbc60c4a57d000bf701057eba0133c